zerver/migrations/0418_backfill_message_realm.py

```diff
@@ -0,0 +1,101 @@
+from django.conf import settings
+from django.db import migrations, transaction
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Max, OuterRef, Subquery
+
+
+def backfill_message_realm(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    RECIPIENT_PERSONAL = 1
+    RECIPIENT_STREAM = 2
+    RECIPIENT_HUDDLE = 3
+
+    Message = apps.get_model("zerver", "Message")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+    Recipient = apps.get_model("zerver", "Recipient")
+    UserProfile = apps.get_model("zerver", "UserProfile")
+
+    BATCH_SIZE = 10000
```

Comment:
Should the BATCH_SIZE be larger perhaps?

Reply:
I don't think so; doing bulk updates on more than 10k rows risks being pretty slow.